frontend/src/scenes/error-tracking/queries.ts

```diff
@@ -21,6 +21,7 @@ export const errorTrackingQuery = ({
             select: [
                 'any(properties) -- Error',
                 'properties.$exception_type',
+                'sparkline(reverse(arrayMap(x -> countEqual(groupArray(dateDiff("hour", now() - INTERVAL 1 day, timestamp)), x), range(24)))) -- Volume',
```

Comment:
@pauldambra I ended up rewriting the query because you made a great point in https://github.com/PostHog/posthog/pull/23160/commits/e128cce253a348dbc8af24e03cfc3f8ec2389e59#r1655641858...

`toHour(timestamp)` gave the hour the event happened which might incorrectly attribute values for events that happen in the same hour but on different days e.g. if the date range is 2 days and two events happen at 6pm but on different days that would end up being `[6,6]` which would get grouped together as `[0, 0, 0, 0, 0, 2, 0, 0, ... (0's up to the 24th index)]` (2 represents the count of the errors that happened in the hour at that index).

---

The right thing to do is a relative date match:
```
dateDiff('hour', timestamp, now())
```
This will give us the number of hours since now() that the event occurred.

The `groupArray` makes an array of the values for every event in the selection, so you end up with something like `[0,0,1,2,3,4,24,3,48,2,70]` which represent the number of hours since each individual exception occurred.

The `range(24)` can be thought of as an empty array with a single value for every hour in the day. For the first version I'm only going to show results from the last day grouped by hour (hence the 24) but we might want to extend that in time.

We use `arrayMap` to loop over the 24 items in the array. `countEqual` finds the number of events that happened in that hour of the 24 by summing the occurrences of that hour (variable `x`) in the grouped array.

Because we want the more recent events to appear later (e.g. events that happened 0 hours ago at the end of the sparkline) we need to reverse the result.

`sparkline` is then just a visual wrapper function to display it nicely in the data table on the frontend

Reply:
awesome explanation, thank you!